src/download.rs

```diff
@@ -79,6 +80,62 @@ impl TryFrom<&str> for Download {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum Status {
+    Fail(String),
+    Success,
+}
+/// Represents a [`Download`] summary.
+#[derive(Debug, Clone)]
+pub struct Summary {
+    /// Downloaded items.
+    download: Download,
+    /// HTTP status code.
+    statuscode: StatusCode,
+    /// Download size in bytes.
+    size: u64,
+    /// Status.
+    status: Status,
+}
+
+impl Summary {
+    /// Create a new [`Download`] [`Summary`].
+    pub fn new(download: Download, statuscode: StatusCode, size: u64) -> Self {
+        Self {
+            download,
+            statuscode,
+            size,
+            status: Status::Fail("Unstarted".into()),
```

Comment:
Should there be a `Status::Uknown` rather than default to a fail state when calling `Summary::new`?

Reply:
I wanted to avoid adding an unknown state when I did this, but I think you are right, it makes more sense to do so 👍

Reply:
Typically it's always good to have an uninitialized enum be a null/unkown variant that way calling `SomeEnum::default()` doesn't accidentally imply incorrect semantics. You can also make the status `Option<Status>`